src/app/loadout-builder/mod-utils.ts

```diff
@@ -30,3 +40,296 @@ export function someModHasEnergyRequirement(mods: PluggableInventoryItemDefiniti
     (mod) => !mod.plug.energyCost || mod.plug.energyCost.energyType !== DestinyEnergyType.Any
   );
 }
+
+function stringifyMods(permutation: (PluggableInventoryItemDefinition | null)[]) {
+  let permutationString = '';
+  for (const modOrNull of permutation) {
+    if (modOrNull) {
+      const energy = modOrNull.plug.energyCost;
+      permutationString += `(${energy?.energyType},${energy?.energyCost},${modOrNull.plug.plugCategoryHash})`;
```

Comment:
I think we only need the `${modOrNull.plug.plugCategoryHash})` for raid mods but I haven't convinced myself entirely of that yet. The idea of this is to create unique permutations of mods for checking. 

For raid mods we need to ensure that we get permutations of mods where the category matters as they can only slot into the associated raid socket. 

While combat mods it might be better to leave this out as we dont care if it's a charged with light mod or a warmind cell mod, we just care about the element type and cost I think.